lib/context.js

```diff
@@ -48,7 +48,11 @@ class FlowContext {
 
   leave () {
     debug('leave')
-    delete this.ctx.session._flow
+    const handler = this.current.leaveMiddleware
+      ? this.current.leaveMiddleware()
+      : this.current.middleware()
```

Comment:
What reason to call current scene middleware?

I think `safePassThru` is more suitable for that:
```js
const handler = this.current.leaveMiddleware
  ? this.current.leaveMiddleware()
  : safePassThru()
```